### ISSUE TITLE
[api-extractor] Naming conflict between import and global type renames global

### ISSUE THREAD

**Issue by rudolf** [microsoft/rushstack#1316]
If there's two type references, one a named import, the other a global variable, api-extractor renames the global variable instead of renaming the named import.

Example:
```typescript
import {React} from 'react';
export interface UsingNamedImport {
  children: React.ReactNode;
}

export interface UsingGlobal {
  children: React_2.ReactNode
}
```

**Comment by octogonz** [comment]
The problem is with `_collectGlobalNames()` which is used by `_makeUniqueNames()`:
https://github.com/microsoft/web-build-tools/blob/d061216478054c8e2011bc3f4a1954b165f48264/apps/api-extractor/src/collector/Collector.ts#L386-L397

The above code aims to detect all names that are defined in the global namespace, but at the time we were unsure how to implement it.  (This is tracked by https://github.com/Microsoft/web-build-tools/issues/1095.)  The proposal was to use internal compiler APIs like this:

```ts
const typeChecker = (this.program as any).getDiagnosticsProducingTypeChecker();
const resolver = (typeChecker as any).getEmitResolver();
resolver.hasGlobalName('Promise'); // --> true
```

However, the comments say that this produced 1,650 results which is a large set that is likely going to have unnecessary matches:  For example, if we implement it that way, it would fix @skaapgif's example above.  But the imported `React` would still get renamed even if the `React` global variable is not used.

Thus perhaps `_collectGlobalNames()` should /only/ return the global names that are actually referenced somewhere in the `Collector`'s API surface.

@rbuckton Do you happen to know if there's an API similar to `resolver.hasGlobalName()` that can check a symbol to see whether it is referring to a global variable?

**Comment by rbuckton** [comment]
There's no public API for this currently, but [a workaround](https://github.com/rbuckton/typedoc-plugin-biblio/blob/master/src/plugin.ts#L57) is possible using `checker.getSymbolsInScope`

**Comment by octogonz** [comment]
Fixed by PR #1767 which was published with API Extractor 7.7.12.

**Comment by octogonz** [comment]
BTW I've also improved `--diagnostics` switch to report when globals are encountered, which will show the path of which file is referencing the `React` global.
